crypto/bn/bn_prime.c

```diff
@@ -363,7 +364,7 @@ int bn_probable_prime_dh(BIGNUM *rnd, int bits,
     if ((t1 = BN_CTX_get(ctx)) == NULL)
         goto err;
 
-    if (!BN_priv_rand(rnd, bits, BN_RAND_TOP_ONE, BN_RAND_BOTTOM_ODD))
+    if (!BN_rand(rnd, bits, BN_RAND_TOP_ONE, BN_RAND_BOTTOM_ODD))
```

Comment:
Why do you use the public drbg in `bn_probable_prime_dh()`and the private drbg in `probable_prime()`?

Reply:
That doesn't make sense at all ...

Reply:
(And clearly why the other one is marked as XXX/TODO)

Reply:
Oh, but this is about primes for use in DH, where the prime is actually send to the peer.